phnt/include/ntkeapi.h

```diff
@@ -138,7 +138,7 @@ NtCallbackReturn(
 
 #if (PHNT_VERSION >= PHNT_VISTA)
 NTSYSCALLAPI
-VOID
+NTSTATUS
```

Comment:
VOID is the correct type for this function and the ntoskrnl disassembly shows the function doesn't return:
```
void NtFlushProcessWriteBuffers()
{
  KeFlushProcessWriteBuffers(0);
}
```

Kernelbase also exports a function named [FlushProcessWriteBuffers](https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-flushprocesswritebuffers) which is declared with VOID.

The export entry is a forwarder alias to `NTDLL.NtFlushProcessWriteBuffers` which can only be done when the signature is identical, so technically NtFlushProcessWriteBuffers is documented on MSDN with a void return type.

Reply:
ntoskrnl.exe 10.0.19041

asm:
```asm
;x64
.text:0000000140272060 NtFlushProcessWriteBuffers proc near
.text:0000000140272060                 sub     rsp, 28h
.text:0000000140272064                 xor     ecx, ecx
.text:0000000140272066                 call    KeFlushProcessWriteBuffers
.text:000000014027206B                 xor     eax, eax ; <<<<<<<<< this
.text:000000014027206D                 add     rsp, 28h
.text:0000000140272071                 retn
.text:0000000140272072                 db 0CCh
.text:0000000140272072 NtFlushProcessWriteBuffers endp

;x86
.text:0052FB4A _NtFlushProcessWriteBuffers@0 proc near
.text:0052FB4A                 xor     cl, cl
.text:0052FB4C                 call    _KeFlushProcessWriteBuffers@4
.text:0052FB51                 xor     eax, eax ; <<<<<<<<< this
.text:0052FB53                 retn
.text:0052FB53 _NtFlushProcessWriteBuffers@0 endp
```

disasm:
```C
// x64
NTSTATUS __stdcall NtFlushProcessWriteBuffers()
{
  KeFlushProcessWriteBuffers(0i64);
  return 0;
}

// x86
NTSTATUS __stdcall NtFlushProcessWriteBuffers()
{
  KeFlushProcessWriteBuffers(0);
  return 0;
}
```

Reply:
Microsoft leaked the type and it's defined NTSTATUS... so I was wrong and you're correct it does return the status 👍